### PR TITLE
Create a .NET maintainers team

### PR DESCRIPTION
**Pull request by mdarocha**
###### Description of changes

This would allow getting help with building .NET packages easier, as well as allows interested maintainers to get pinged about updates to the core .NET infrastructure

I added the initial members based on recent github activity and `git shortlog`. Comment on this PR if you would like to be added or removed - it shouldn't without everybody agreeing to be part of this team.

I also updated CODEOWNERS to not only contain @IvarWithoutBones, but all members of this dotnet team.

### CODE REVIEW

**Review by JamieMagee — APPROVED** [review]

File: .github/CODEOWNERS
Comment (fricklerhandwerk) [inline .github/CODEOWNERS:294]:
```suggestion
/pkgs/test/dotnet                   @IvarWithoutBones @mdarocha @corngood @raphaelr @JamieMagee
/doc/languages-frameworks/dotnet.section.md @IvarWithoutBones @mdarocha @corngood @raphaelr @JamieMagee
```

Please don't forget to also maintain related documentation. @NixOS/documentation-team is glad to assist with global questions.

Comment (fricklerhandwerk) [reply .github/CODEOWNERS:294]: Also, wouldn't it be easier to have a GitHub team and manage membership there? At least it would be a single source of truth.

Comment (fricklerhandwerk) [reply .github/CODEOWNERS:294]: > I assumed the teams.nix file creates github teams automatically?

That would very much surprise me, but I have no idea about how this works behind the scenes.

Comment (RaitoBezarius) [reply .github/CODEOWNERS:294]: It doesn't, the teams.nix is extra metadata for various Nixpkgs scripts. You still need to create the team separately.

**Review by corngood — APPROVED** [review]

**Review by raphaelr — APPROVED** [review]

File: .github/CODEOWNERS
Comment (fricklerhandwerk) [reply .github/CODEOWNERS]: > I believe commit access is handled per-contributor.

It *could* be managed by giving a maintainer team write access and then having someone responsible to curate team membership. There is no rule in place at the moment, as far as I know.

**Comment by felschr** [comment]
Should all members be added here as well?
https://github.com/NixOS/nixpkgs/blob/73b17bf327953b64c0acfe6cead40a7ffc4fd41f/pkgs/development/compilers/dotnet/build-dotnet.nix#L163

File: doc/languages-frameworks/dotnet.section.md
Comment (fricklerhandwerk) [inline doc/languages-frameworks/dotnet.section.md:157]:
```suggestion
When packaging a new .NET application in nixpkgs, you can tag the [`@NixOS/dotnet`](https://github.com/orgs/nixos/teams/dotnet) team for help and code review.
```

Merge only once the team exists.

**Comment by 06kellyjac** [comment]
IMO it'd be worth adding members to the team now so that the docs are accurate as soon as the PR is merged & CODEOWNERS can also be set to the team rather than in a second PR

File: .github/CODEOWNERS
Comment (RaitoBezarius) [reply .github/CODEOWNERS]: > Maybe @RaitoBezarius or somebody else can shine more light on this, but I don't think any NixOS team have write access? I believe commit access is handled per-contributor.

Yes, I'm not aware of that, do you really need the constant pinging for everything in that tree? If not, consider waiting to have enough work to ask for write access and add only the active members of the .NET in that file.

Comment (RaitoBezarius) [reply .github/CODEOWNERS]: Got it. :)

**Comment by roberth** [comment]
> Can this PR be merged? Or is there something more that needs to be done here?

Nothing that's been listed [in the docs](https://nixos.org/manual/nixpkgs/stable/#reviewing-contributions-maintainer-teams), so lgtm!

I can't promise that the doc is exhaustive, but if not, we can fix it :muscle:

**Pull request merged**
